include/smack/smack.h

```diff
@@ -57,49 +65,271 @@ int __SMACK_nondet() {
+  D("axiom $and.i1(1,1) == 1;");
+  D("function $or.i1(p1:int, p2:int) returns (int);");
+  D("axiom $or.i1(0,0) == 0;");
+  D("axiom $or.i1(0,1) == 1;");
+  D("axiom $or.i1(1,0) == 1;");
+  D("axiom $or.i1(1,1) == 1;");
+  D("function $xor.i1(p1:int, p2:int) returns (int);");
+  D("axiom $xor.i1(0,0) == 0;");
+  D("axiom $xor.i1(0,1) == 1;");
+  D("axiom $xor.i1(1,0) == 1;");
+  D("axiom $xor.i1(1,1) == 0;");
+
+  // Predicates
+  D("function {:inline} $ult.i64(p1:i64, p2:i64) returns (bool) {p1 < p2}");
+  D("function {:inline} $ugt.i64(p1:i64, p2:i64) returns (bool) {p1 > p2}");
+  D("function {:inline} $ule.i64(p1:i64, p2:i64) returns (bool) {p1 <= p2}");
+  D("function {:inline} $uge.i64(p1:i64, p2:i64) returns (bool) {p1 >= p2}");
+  D("function {:inline} $slt.i64(p1:i64, p2:i64) returns (bool) {p1 < p2}");
+  D("function {:inline} $sgt.i64(p1:i64, p2:i64) returns (bool) {p1 > p2}");
+  D("function {:inline} $sle.i64(p1:i64, p2:i64) returns (bool) {p1 <= p2}");
+  D("function {:inline} $sge.i64(p1:i64, p2:i64) returns (bool) {p1 >= p2}");
+
+  D("function {:inline} $ult.i32(p1:i32, p2:i32) returns (bool) {p1 < p2}");
+  D("function {:inline} $ugt.i32(p1:i32, p2:i32) returns (bool) {p1 > p2}");
+  D("function {:inline} $ule.i32(p1:i32, p2:i32) returns (bool) {p1 <= p2}");
+  D("function {:inline} $uge.i32(p1:i32, p2:i32) returns (bool) {p1 >= p2}");
+  D("function {:inline} $slt.i32(p1:i32, p2:i32) returns (bool) {p1 < p2}");
+  D("function {:inline} $sgt.i32(p1:i32, p2:i32) returns (bool) {p1 > p2}");
+  D("function {:inline} $sle.i32(p1:i32, p2:i32) returns (bool) {p1 <= p2}");
+  D("function {:inline} $sge.i32(p1:i32, p2:i32) returns (bool) {p1 >= p2}");
+
+  D("function {:inline} $ult.i16(p1:i16, p2:i16) returns (bool) {p1 < p2}");
+  D("function {:inline} $ugt.i16(p1:i16, p2:i16) returns (bool) {p1 > p2}");
+  D("function {:inline} $ule.i16(p1:i16, p2:i16) returns (bool) {p1 <= p2}");
+  D("function {:inline} $uge.i16(p1:i16, p2:i16) returns (bool) {p1 >= p2}");
+  D("function {:inline} $slt.i16(p1:i16, p2:i16) returns (bool) {p1 < p2}");
+  D("function {:inline} $sgt.i16(p1:i16, p2:i16) returns (bool) {p1 > p2}");
+  D("function {:inline} $sle.i16(p1:i16, p2:i16) returns (bool) {p1 <= p2}");
+  D("function {:inline} $sge.i16(p1:i16, p2:i16) returns (bool) {p1 >= p2}");
+
+  D("function {:inline} $ult.i8(p1:i8, p2:i8) returns (bool) {p1 < p2}");
+  D("function {:inline} $ugt.i8(p1:i8, p2:i8) returns (bool) {p1 > p2}");
+  D("function {:inline} $ule.i8(p1:i8, p2:i8) returns (bool) {p1 <= p2}");
+  D("function {:inline} $uge.i8(p1:i8, p2:i8) returns (bool) {p1 >= p2}");
+  D("function {:inline} $slt.i8(p1:i8, p2:i8) returns (bool) {p1 < p2}");
+  D("function {:inline} $sgt.i8(p1:i8, p2:i8) returns (bool) {p1 > p2}");
+  D("function {:inline} $sle.i8(p1:i8, p2:i8) returns (bool) {p1 <= p2}");
+  D("function {:inline} $sge.i8(p1:i8, p2:i8) returns (bool) {p1 >= p2}");
+
+  D("function {:inline} $i2b(i: int) returns (bool) {i != 0}");
```

Comment:
Since `$i2b` and `$b2i` should only be converting between `i1` and `bool`, the argument of `$i2b` and the return of `$b2i` should have types `i1`, not `int` and `i8`.